lib/rubocop/ast/node/casgn_node.rb

```diff
@@ -19,6 +19,7 @@ def namespace
       def name
         node_parts[1]
       end
+      alias short_name name
```

Comment:
`CasgnNode` and `Const` should have a symmetrical interface so that they can be compared (eg. in `Lint/SelfAssignment`).

Reply:
Sounds good, but maybe the best would be to simply move all the implementation of `Const` to a module and include this here too?

Reply:
@marcandre if you can take another look, I have extracted the methods from `ConstNode` and included them in `CasgnNode` as well. I called the mixin `ConstantNode` but happy for a suggestion for a better name if you'd like.

This adds some new methods to `CasgnNode` to mirror `ConstNode`, so I've added tests for those as well.